src/System.DirectoryServices.AccountManagement/tests/ComputerPrincipalTest.cs

```diff
@@ -9,25 +9,72 @@ namespace System.DirectoryServices.AccountManagement.Tests
     public class ComputerPrincipalTest : PrincipalTest
     {
         [Fact]
+        public void Ctor_Context()
+        {
+            var context = new PrincipalContext(ContextType.Machine);
+            var principal = new ComputerPrincipal(context);
+            Assert.Equal(context, principal.Context);
+            Assert.Empty(principal.ServicePrincipalNames);
+        }
+
+        [Fact]
+        public void Ctor_NullContext_ThrowsArgumentException()
+        {
+            AssertExtensions.Throws<ArgumentException>(null, () => new ComputerPrincipal(null));
+            AssertExtensions.Throws<ArgumentException>(null, () => new ComputerPrincipal(null, "samAcountName", "password", enabled: true));
+        }
+
+        [Fact]
+        public void Ctor_NullSamAccountName_ThrowsArgumentException()
+        {
+            var context = new PrincipalContext(ContextType.Machine);
+            AssertExtensions.Throws<ArgumentException>(null, () => new ComputerPrincipal(context, null, "password", enabled: true));
+        }
+
+        [Fact]
+        public void Ctor_EmptySamAccountName_ThrowsArgumentNullException()
+        {
+            var context = new PrincipalContext(ContextType.Machine);
+            AssertExtensions.Throws<ArgumentNullException>("Principal.SamAccountName cannot be null or empty.", () => new ComputerPrincipal(context, string.Empty, "password", enabled: true));
+        }
+
+        [Fact]
+        public void Ctor_NullPassword_ThrowsArgumentException()
+        {
+            var context = new PrincipalContext(ContextType.Machine);
+            AssertExtensions.Throws<ArgumentException>(null, () => new ComputerPrincipal(context, "samAccountName", null, enabled: true));
+        }
+
+        [ConditionalFact(nameof(PlatformDetection) + "." + nameof(PlatformDetection.IsNotWindowsNanoServer))]
+        public void Ctor_MachineContext_ThrowsInvalidCastException()
+        {
+            var context = new PrincipalContext(ContextType.Machine);
+            Assert.Throws<InvalidCastException>(() => new ComputerPrincipal(context, "samAccountName", "password", enabled: true));
+        }
+
+        [Fact]
         public void ComputerPrincipalConstructorTest()
         {
+            if (DomainContext == null)
+            {
+                return;
+            }
```

Comment:
Previously, we'd run the following on initialization: https://github.com/dotnet/corefx/blob/c79de0509785d6d126806c39bd67769efdd280a3/src/System.DirectoryServices.AccountManagement/tests/PrincipalTest.cs#L21-L30

However, this means that every test fails with
```
System.DirectoryServices.AccountManagement.Tests.ComputerPrincipalTest.Ctor_Context [FAIL]
        System.DirectoryServices.AccountManagement.PrincipalServerDownException : The server could not be contacted.
        ---- System.DirectoryServices.Protocols.LdapException : The LDAP server is unavailable.
        Stack Trace:
              at System.DirectoryServices.AccountManagement.PrincipalContext.ReadServerConfig(String serverName, Server
  Properties& properties)
              at System.DirectoryServices.AccountManagement.PrincipalContext.DoServerVerifyAndPropRetrieval()
              at System.DirectoryServices.AccountManagement.PrincipalContext..ctor(ContextType contextType, String name
  , String container, ContextOptions options, String userName, String password)
              at System.DirectoryServices.AccountManagement.PrincipalContext..ctor(ContextType contextType, String name
  , String container, String userName, String password)
              at System.DirectoryServices.AccountManagement.Tests.PrincipalTest.RefreshContext()
              at System.DirectoryServices.AccountManagement.Tests.PrincipalTest..ctor()
              at System.DirectoryServices.AccountManagement.Tests.ComputerPrincipalTest..ctor()
           ----- Inner Stack Trace -----
              at System.DirectoryServices.Protocols.LdapConnection.Connect()
              at System.DirectoryServices.Protocols.LdapConnection.SendRequestHelper(DirectoryRequest request, Int32& m
  essageID)
              at System.DirectoryServices.Protocols.LdapConnection.SendRequest(DirectoryRequest request, TimeSpan reque
  stTimeout)
              at System.DirectoryServices.Protocols.LdapConnection.SendRequest(DirectoryRequest request)
              at System.DirectoryServices.AccountManagement.PrincipalContext.ReadServerConfig(String serverName, Server
  Properties& properties)
```

So what I did was surround the throwing code with a catch. This allows the tests to run, but leaves `DomainContext` uninitialized. So inside the test methods that rely on `DomainContext`, I've checked if it's uninitialized or not.

The reason why these tests never failed before is because they've never been run before. The `configuration.props` file was entirely empty so the tests never actually ran. This caused issues like #21496 to go unoticed